esphome/components/ethernet/ethernet_component.cpp

```diff
@@ -115,7 +121,19 @@ void EthernetComponent::setup() {
 
   eth_w5500_config_t w5500_config = ETH_W5500_DEFAULT_CONFIG(spi_handle);
 #endif
-  w5500_config.int_gpio_num = this->interrupt_pin_;
+  if (this->interrupt_pin_ >= 0) {
+    w5500_config.int_gpio_num = this->interrupt_pin_;
+#ifdef ETHERNET_SPI_WITHOUT_IRQ_AVAILABLE
+    w5500_config.poll_period_ms = 0;
+#endif
+  } else {
+#ifdef ETHERNET_SPI_WITHOUT_IRQ_AVAILABLE
+    w5500_config.int_gpio_num = -1;
+    w5500_config.poll_period_ms = this->polling_interval_;
+#else
+    ESP_LOGE(TAG, "IRQ pin number is negative(%d), seems to be configuration error", this->interrupt_pin_);
+#endif
+  }
```

Comment:
Given the validations on the python side, and the defaults in the header, I don't think you need to do super complex logic here at all
```suggestion
  w5500_config.int_gpio_num = this->interrupt_pin_;
#ifdef USE_ETHERNET_SPI_POLLING_SUPPORTED
  w5500_config.poll_period_ms = this->polling_interval_;
#endif
```

Reply:
Resolved in commit [92fa95d](https://github.com/esphome/esphome/pull/7503/commits/92fa95d180eb749d7680cb3d5faaf567312ebf8a).